quinn-proto/src/lib.rs

```diff
@@ -176,7 +176,8 @@ impl fmt::Display for StreamId {
 }
 
 impl StreamId {
-    pub(crate) fn new(initiator: Side, dir: Dir, index: u64) -> Self {
+    #[doc(hidden)]
```

Comment:
@Ralith you okay with this? Should we attach explicit comments to some of these `#[doc(hidden)]` to clarify to readers that we don't consider these to be part of the public API?

Reply:
There are no public functions that accept a `StreamId` in `quinn`, so I guess it does not open any unwanted API usage at this level. This is not the case of `quinn-proto`.

We could add a new type that is allowed to be publicly constructed, but not accepted by the API ?

Reply:
Sorry, missed the ping somehow! If I understand correctly, this is motivated by HTTP/3 passing Stream IDs explicitly at the application protocol layer. Is that correct? If so, I think that's evidence that maybe this should actually be public for the benefit of application protocol implementers in general. There's no specific reason I'm aware of that this has to be private except the general principle of "don't export things without a need." If downstream code constructs nonsense Stream IDs and gets nonsense behavior, that's on them.

Reply:
This is motivated by the need of calculating a correct `StreamId` raw value to be used in `GoAway` frames and the shutdown mechanism.

All right, we're all set then ! Thanks !

Reply:
I think @Ralith was suggesting to make this properly public, so please remove `#[doc(hidden)]` and add a short docstring.